apps/src/code-studio/progress.js

```diff
@@ -165,9 +165,12 @@ progress.renderCourseProgress = function(scriptData) {
   initViewAs(store, scriptData);
   queryUserProgress(store, scriptData, null);
 
-  const teacherResources = (scriptData.teacher_resources || []).map(
-    ([type, link]) => ({type, link})
-  );
+  const teacherResources = scriptData.is_migrated
+    ? scriptData.teacher_resources
+    : (scriptData.teacher_resources || []).map(([type, link]) => ({
+        type,
+        link
+      }));
```

Comment:
storing different data types in the same field scares me a bit. could these be stored in separate fields? using the same field is (for me) harder to reason about, and it seems like using the wrong value of is_migrated could lead to a harder-to-debug type mismatch. I'm not 100% sure but I think this might have made it a bit easier to split up the PR, too.

Even if that sounds like a better solution to you, this is definitely a judgment call on whether it's worth doing... this code has an expiration date on it, so some added complexity could be ok, but on the other hand that date is still a couple months out. Happy to discuss :-)

Reply:
hmm let me try that out. I think I was trying to avoid too much `migratedTeacherResources` kinds of things but with the two resource editors, that didn't really work out anyway

Reply:
Took a stab at this in fcfcac9ad5a. Does this seem better? Part of me thinks it makes reasoning about the props simpler, like you said, but also the logic about which resources to use seems to leak through more. Thoughts?

Reply:
thank you for trying this out! personally, having the logic "leak through" more seems to make the code a lot more readable to me. I would vote for this approach. Are there particular parts of this solution that are most troublesome to you?

Reply:
Not particularly troublesome. If it's clearer, then I think it's a better approach. I'll update tests to match